Source/CarthageKit/Xcode.swift

```diff
@@ -1080,6 +1082,17 @@ public func architecturesInPackage(_ packageURL: URL) -> SignalProducer<String,
 		}
 }
 
+/// Strips debug symbols from the given framework
+public func stripDebugSymbols(_ frameworkURL: URL) -> SignalProducer<(), CarthageError> {
+	return SignalProducer<URL, CarthageError> { () -> Result<URL, CarthageError> in binaryURL(frameworkURL) }
+		.flatMap(.merge) { binaryURL -> SignalProducer<TaskEvent<Data>, CarthageError> in
+			let stripTask = Task("/usr/bin/xcrun", arguments: [ "strip", "-S","-o", binaryURL.path, binaryURL.path])
```

Comment:
is not -x needed?

```
 -S     Remove the debugging symbol table entries (those created by the -g option to cc(1) and other
              compilers).
 -x     Remove all local symbols (saving only global symbols).
```

Reply:
So to figure out what flags to pass, I reverse-engineered Xcode by replacing the `strip` binary on my system with one that logs all the arguments passed to it. Xcode passes this flag and only this flag when using a vanilla copy. So I figured we should be doing what Xcode does for symmetry.

I am by far not an expert on the intricacies of stripping symbols, so I'm happy to listen to alternatives however.

Reply:
Further info... there is a setting STRIP_STYLE which seems to affect this. Default for me was "Debugging Symbols", but "Non-Global Symbols" seems to be what brings in "-x".

The documentation for this property seems to indicate it applies only to the linked product and not to strip on copy, but I'm suspicious that this is not the case. Will investigate further.

Reply:
Maybe then it should respect STRIP_STYLE ? I'm not an expert either so I guess that respecting the style is safe bet.

Reply:
Decision was made to leave -S only.

See:
- https://github.com/Carthage/Carthage/pull/2361#issuecomment-370429346
- https://github.com/Carthage/Carthage/pull/2361#issuecomment-370436508